src/Simplex/Chat.hs

```diff
@@ -660,6 +675,33 @@ processChatCommand = \case
       let s = connStatus $ activeConn (ct :: Contact)
        in s == ConnReady || s == ConnSndReady
 
+-- mobile clients use file paths relative to app directory (e.g. for the reason ios app directory changes on updates),
+-- so we have to differentiate between the file path stored in db and communicated with frontend, and the file path
+-- used during file transfer for actual operations with file system
+toFSFilePath :: ChatMonad m => FilePath -> m FilePath
+toFSFilePath f = do
+  ff <- asks filesFolder
+  readTVarIO ff >>= \case
+    Nothing -> pure f
+    Just filesFolder -> pure $ filesFolder <> "/" <> f
+
+deleteFiles :: ChatMonad m => [(Int64, Maybe FilePath, ACIFileStatus)] -> m ()
+deleteFiles files = do
+  ff <- asks filesFolder
+  readTVarIO ff >>= \case
+    Nothing -> pure () -- only delete files if filesFolder is set (i.e. on mobile devices)
+    Just filesFolder ->
+      forM_ files $ \(fileId, filePath_, status) -> do
+        case status of
+          AFS _ CIFSRcvTransfer -> closeFileHandle fileId rcvFiles
+          _ -> pure ()
+        case filePath_ of
+          Just filePath -> do
+            let fsFilePath = filesFolder <> "/" <> filePath
+            removeFile fsFilePath `E.catch` \(_ :: E.SomeException) ->
+              removePathForcibly fsFilePath `E.catch` \(_ :: E.SomeException) -> pure ()
```

Comment:
why is it needed?

Reply:
just in case file couldn't be deleted normally
https://hackage.haskell.org/package/directory-1.3.7.0/docs/System-Directory.html#v:removePathForcibly
e.g. if for some reason the file permissions were changed to read only it would still attempt to remove it.